c/datasetService.c

```diff
@@ -84,72 +84,136 @@ static int serveDatasetMetadata(HttpService *service, HttpResponse *response) {
   return 0;
 }
 
-static int serveDatasetContents(HttpService *service, HttpResponse *response){
+static int serveVSAMDatasetContents(HttpService *service, HttpResponse *response){
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "begin %s\n", __FUNCTION__);
   HttpRequest *request = response->request;
-
+  
+  serveVSAMCache *cache = (serveVSAMCache *)service->userPointer;
   if (!strcmp(request->method, methodGET)) {
-    char *l1 = stringListPrint(request->parsedFile, 1, 1, "/", 0);
-    char *percentDecoded = cleanURLParamValue(response->slh, l1);
-    char *filenamep1 = stringConcatenate(response->slh, "//'", percentDecoded);
-    char *filename = stringConcatenate(response->slh, filenamep1, "'");
+    char *filename = stringListPrint(request->parsedFile, 2, 1, "/", 0);
     zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_INFO, "Serving: %s\n", filename);
     fflush(stdout);
-    respondWithDataset(response, filename, TRUE);
+    respondWithVSAMDataset(response, filename, cache->acbTable, TRUE);
   }
-  else if (!strcmp(request->method, methodPOST)){
-    char *l1 = stringListPrint(request->parsedFile, 1, 1, "/", 0);
-    char *percentDecoded = cleanURLParamValue(response->slh, l1);
-    char *filenamep1 = stringConcatenate(response->slh, "//'", percentDecoded);
-    char *filename = stringConcatenate(response->slh, filenamep1, "'");
+  else if (!strcmp(request->method, methodPUT)){
+    char *filename = stringListPrint(request->parsedFile, 2, 1, "/", 0);
     zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_INFO, "Updating if exists: %s\n", filename);
     fflush(stdout);
-    updateDataset(response, filename, TRUE);
+    updateVSAMDataset(response, filename, cache->acbTable, TRUE);
+  }
+  else{
+    jsonPrinter *out = respondWithJsonPrinter(response);
 
+    setContentType(response, "text/json");
+    setResponseStatus(response, 405, "Method Not Allowed");
+    addStringHeader(response, "Server", "jdmfws");
+    addStringHeader(response, "Transfer-Encoding", "chunked");
+    addStringHeader(response, "Allow", "GET, PUT");
+    writeHeader(response);
+
+    jsonStart(out);
+    jsonEnd(out);
+
+    finishResponse(response);
   }
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "end %s\n", __FUNCTION__);
-  zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_INFO, "Returning from servedatasetcontents\n");
+  zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_INFO, "Returning from serveVSAMdatasetcontents\n");
   return 0;
 }
 
-static int serveVSAMDatasetContents(HttpService *service, HttpResponse *response){
+static int serveDataset(HttpService *service, HttpResponse *response){
   zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "begin %s\n", __FUNCTION__);
   HttpRequest *request = response->request;
-  serveVSAMCache *cache = (serveVSAMCache *)service->userPointer;
+  char *l1 = stringListPrint(request->parsedFile, 2, 1, "/", 0);
```

Comment:
I really wish this was put into a function or something. All you are doing here is checking to see if there is a member in the request, right? Just make a function to do this, at it could be useful at some point.

bool isMember = checkForDatasetMember(args)

something like this.